### ISSUE TITLE
Rewrite paths in markdown for in-browser contents

### ISSUE THREAD

**Issue by bollwyvl** [jupyterlite/jupyterlite#271]
### Problem

Consider a published JupyterLite site:

```
{lite_dir}/
  _output/              # from here is hosted (perhaps with a prefix) on e.g. _RTD_
    files/
      train.jpg*        # * this may or _may not_ exist
      README.md
      untitled.ipynb**  # ** may or _may not_ be able to generate `train.jpg`
```

Presently, if `README.md` is like the [GFM example](https://github.github.com/gfm/#example-584):

```md
![foo *bar*][]

[foo *bar*]: train.jpg "train & tracks")
```

Opening a Markdown preview of `README.md` (a la #139) would:
- [x]  :tada: **succeed** in loading e.g. `/files/train.jpg`, if it _does_ exist on the server , but...
- [ ] **fail** to load if a user had _added_ such a file by upload #119,/#91 or...
- [ ] **fail** to load if such a file if it was created as the result of running `untitled.ipynb`, or...
- [ ] **fail** to update in response to the file changing on the server, kernel, or browser-based storage
  - > this is _also_ an upstream limitation

### Proposed Solution

Ignoring the last two of the failures above, and just concentrating on the in-brower storage, we'd probably need to:
- intercept all of the `src` and `href` tags that would resolve to `/files`
- see if we have them in the local Contents store (a la `localForage`)
- if we do, rewrite them as `Blob` links
  - this will probably require some caching/cache invalidation scheme

### Additional context

Some markdown ideas:

- hook the Markdown renderer
  - extend the [core renderer](https://github.com/jupyterlab/jupyterlab/blob/a2d2c5b26f16b97e605b43e47cfb9a301620921d/packages/rendermime/src/renderers.ts#L618)
  - replace the core renderer
  - adopt [jupyterlab-markup](https://github.com/agoose77/jupyterlab-markup)
    - and/or forthcoming `jupyterlab-myst` or whatever

### Future Work

- figure out how to make this work for kernel-created resources

**Comment by bollwyvl** [comment]
After #655, this would be more interesting, as kernels would be able to generate files directly into contents that can be viewed in markdown.
